### ISSUE TITLE
Deleted Container Reappears On Devices Even after Syncing After Deletion

### ISSUE THREAD

**Issue by levski** [mozilla/multi-account-containers#2682]
### Before submitting a bug report

- [X] I updated to the latest version of Multi-Account Container and tested if I can reproduce the issue
- [X] I searched for existing reports to see if it hasn't already been reported

### Step to reproduce

open tab for container 1, open website that was once set to open for container 2 only (in my case it was outlook.com)
it will then ask you which container, you should open the website in (which shouldn't appear since the other container was deleted in the past)

### Actual behavior

randomly, the extension will sync even deleted containers, when it does this issue will happen.

### Expected behavior

syncing extension should sync to the master list and sync to all devices.

### Additional informations

not sure what else to say here.

**Comment by StormPooper** [comment]
I've also had deleted containers reappearing recently.

**Comment by newptcai** [comment]
Yes. Deleting container does not seem to work.

**Comment by wondercomm** [comment]
Same issue here. Is there a fix for this yet?

**Comment by krc** [comment]
Possibly related to or same as #2717 (newer), unclear if that relates to syncing. In my case at least I do have syncing turned on. Would be interesting to hear of a case with no sync.

**Comment by alexnt85** [comment]
> Possibly related to or same as [#2717](https://github.com/mozilla/multi-account-containers/issues/2717) (newer), unclear if that relates to syncing. In my case at least I do have syncing turned on. Would be interesting to hear of a case with no sync.

I recently migrated my firefox profile to waterfox and I'm facing this issue. There is a specific container called "private" that reappears and I don't have sync enabled.

**Comment by hovancik** [comment]
having same issue

**Comment by wondercomm** [comment]
> I had a collection of bookmarks in a Container. I don't need the Container anymore. So I deleted the Container (i.e. Remove). I sync all devices and the Container is gone. A few days later, the Container that I deleted is back. I remove the Container, and sync. Check other devices and all OK. Then a few days later, the Container is back. It feels like Groundhog Day ...
> 
> Maybe I have to manually delete all the bookmarks in that Container???

I simply renamed the container(s) that kept coming back and used it for other purposes instead of creating new container. This change was persistent and for now this is a workaround until Mozilla fixes this.

**Comment by hovancik** [comment]
So syncing does work for you? I have multiple devices but containers are not synced as I would expect.

**Comment by jacobthemagicalpanda** [comment]
I am having a similar issue. I accidentally set a website to auto open with a particular container. When I undo this, delete the website from the list of associated websites under the manage container settings, it works for a while but eventually comes back and tries to auto open in that container again. Very frustrating.

**Comment by bunk3m** [comment]
> So syncing does work for you? I have multiple devices but containers are not synced as I would expect.

Yes, the sync is working for me.  My comment was only about deleting a container but it always coming back a few days later.  So while it deleted temporarily, it shows up again.

**Comment by bergsj** [comment]
Same problem here since working on two devices with FF and Multi-Containers. Before I was working with Multi-Containers one just one laptop and this was working perfectly. Now it is just super annoying to select the correct container through the clutter....

**Comment by LPS100** [comment]
2026 and same issue, I have 4 containers and there's a stray container from the defaults that constantly keeps ressurecting itself. I've tried to manually delete the container from its json file and nothing. 😕

**Comment by bunk3m** [comment]
This is so annoying.  Is anyone looking at this?  I can't get rid of old containers.  And since they sync, I delete them on one laptop only to find them show up on my desktop.  Delete it on my desktop again, sync, sync.  And wow, back again.  This is soooo close to being a great feature but it's a fail in execution.  :(

**Comment by notDavid** [comment]
This issue is very annoying, i have many (temporary) containers and i cannot get rid of them. This bug has existed for so long now.

Whoever wrote this spaghetti sync code, should be severly punnished: I suggest they never get invited to any Cocktail parties, Barbecues and Slumber parties, for the rest of their life ;-)
